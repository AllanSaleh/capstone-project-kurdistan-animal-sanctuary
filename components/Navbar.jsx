import React, { useState } from "react";
import { useTranslation } from "next-i18next";
import Link from "next/link";
import LoginandLanguage from "./LoginandLanguage";
import MobileMenu from "./MobileMenu";
import Navbutton from "./Navbutton";

const Navbar = () => {
  const [display, setDisplay] = useState(false);
  const { t } = useTranslation("navbar");

  const handleclick = () => {
    if (display === true) {
      setDisplay(false);
    } else {
      setDisplay(true);
    }
  };

  return (
    <div>
      <nav className="flex flex-row-reverse  md:flex-row bg-primary  py-6 px-2 h-20 md:h-24 justify-items-center font-Quicksand">
        <div className="w-2/4  h-14 relative md:h-20">
          <Link href="./">
            <img
              className="w-12 h-12 absolute -top-2 -left-5 md:left-0 cursor-pointer md:w-16 md:h-16"
              src="./Logo.png"
              alt="Website logo "
            />
          </Link>
        </div>
        <div className="flex flex-row  items-center w-2/4 md:justify-around md:items-center">
          <div className="  float-right whitespace-nowrap hidden md:inline-block">
            <span>
              <Navbutton name={t("navbar.Home")} link="./" />

              <Navbutton name={t("navbar.Contact")} link="./ContactUs" />
              <Navbutton name={t("navbar.Adopt")} link="./AdoptThePerfectPet" />
              <Navbutton name={t("navbar.Resources")} link="./Resource" />
            </span>
            <LoginandLanguage />
          </div>

          <div className="inline-block  float-right whitespace-nowrap md:hidden items-center justify-center">
            <button
              type="button"
              className=" w-full h-full font-bold rounded-lg    text-secondary flex "
              onClick={() => {
                handleclick();
              }}
            >
              <svg
                xmlns="http://www.w3.org/2000/svg"
                className="h-12 w-12"
                fill="none"
                viewBox="0 0 24 24"
                stroke="currentColor"
              >
                <path d="M4 6h16M4 12h16M4 18h16" />
              </svg>
            </button>
          </div>
        </div>
      </nav>
      <MobileMenu state={display} />
    </div>
  );
};
export default Navbar;
