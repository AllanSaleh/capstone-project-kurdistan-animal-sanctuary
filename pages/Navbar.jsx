import React from "react";
import Link from "next/link";
import Navbutton from "./Navbutton";

const Navbar = () => (
  <nav className="flex flex-row  bg-primary  py-6 px-24 h-24 justify-items-center ">
    {" "}
    <div className="w-2/4 h-20 relative">
      {" "}
      <Link href="./">
        <img
          className="w-16 h-16 absolute -top-2"
          src="./Logo.png"
          alt="animal profile "
        />
      </Link>{" "}
    </div>
    <div className="flex flex-row justify-around items-center  w-2/4">
      {" "}
      <div>
        <Navbutton name="Home" link="./" />
        <Navbutton name="About" link="./" />
        <Navbutton name="Contact" link="./" />
        <Navbutton name="Adopt" link="./" />
        <Navbutton name="Resources" link="./" />
      </div>{" "}
      <div className="pr-8">
        {" "}
        <Link href="/">
          <button
            type="button"
            className="mr-3.5  w-28 h-10 text-highlight border-2 border-highlight rounded-lg "
          >
            {" "}
            Login
          </button>
        </Link>
        <Link href="/">
          <button
            type="button"
            className="mr-3 w-28 h-10 text-primary bg-highlight rounded-lg "
          >
            {" "}
            Sign up
          </button>
        </Link>
      </div>
    </div>
  </nav>
);
export default Navbar;
