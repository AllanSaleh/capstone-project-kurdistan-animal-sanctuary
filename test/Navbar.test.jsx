import React from "react";
import renderer from "react-test-renderer";
import { test, expect } from "@jest/globals";
import Navbar from "../components/Navbar";

test("proparly renders the AnimalOfWeek component", () => {
  const tree = renderer.create(<Navbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
