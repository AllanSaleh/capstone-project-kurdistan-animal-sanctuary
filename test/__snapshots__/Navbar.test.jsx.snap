// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proparly renders the Navbar component 1`] = `
<div>
  <nav
    className="flex flex-row-reverse  md:flex-row bg-primary  py-6 px-2 h-20 md:h-24 justify-items-center font-Quicksand"
  >
    <div
      className="w-2/4  h-14 relative md:h-20"
    >
      <img
        alt="Website logo "
        className="w-12 h-12 absolute -top-2 -left-5 md:left-0 cursor-pointer md:w-16 md:h-16"
        onClick={[Function]}
        onMouseEnter={[Function]}
        src="./Logo.png"
      />
    </div>
    <div
      className="flex flex-row  items-center w-2/4 md:justify-around md:items-center"
    >
      <div
        className="  float-right whitespace-nowrap hidden md:inline-block"
      >
        <span>
          <button
            className="mr-3 text-gray-300 hover:text-highlight"
            onClick={[Function]}
            onMouseEnter={[Function]}
            type="button"
          >
            Home
          </button>
          <button
            className="mr-3 text-gray-300 hover:text-highlight"
            onClick={[Function]}
            onMouseEnter={[Function]}
            type="button"
          >
            Contact
          </button>
          <button
            className="mr-3 text-gray-300 hover:text-highlight"
            onClick={[Function]}
            onMouseEnter={[Function]}
            type="button"
          >
            Adopt
          </button>
          <button
            className="mr-3 text-gray-300 hover:text-highlight"
            onClick={[Function]}
            onMouseEnter={[Function]}
            type="button"
          >
            Resources
          </button>
        </span>
        <span>
          <span
            className="hidden"
          >
            <button
              onClick={[Function]}
              onMouseEnter={[Function]}
              type="button"
            >
              <img
                alt="Website logo "
                className="w-16 h-16  cursor-pointer  rounded-full   "
                src="./profile.png"
              />
            </button>
          </span>
          <span
            className="block mt-7 sm:inline-block sm:mt-0 "
          >
            <button
              className="mr-3.5 w-28 h-10 text-highlight border-2 border-highlight rounded-lg "
              onClick={[Function]}
              onMouseEnter={[Function]}
              type="button"
            >
              Login
            </button>
            <button
              className=" w-28 h-10 text-primary bg-highlight hover:bg-darken rounded-lg "
              onClick={[Function]}
              onMouseEnter={[Function]}
              type="button"
            >
              Sign up
            </button>
          </span>
          <span
            className="bg-secondary text-sm text-white leading-none  rounded-full inline-flex mt-6 mb-4 md:mb-0 sm:mt-0 ml-24  sm:ml-14"
          >
            <button
              className="inline-flex items-center transition-colors duration-300 ease-in focus:outline-none hover:text-black focus:text-primary foucs:bg-primary rounded-l-full px-4 py-2 active"
              type="button"
            >
              <span>
                en
              </span>
            </button>
            <button
              className="inline-flex items-center transition-colors duration-300 ease-in focus:outline-none hover:text-black focus:text-primary foucs:bg-primary rounded-l-full px-4 py-2"
              type="button"
            >
              <span>
                ku
              </span>
            </button>
          </span>
        </span>
      </div>
      <div
        className="inline-block  float-right whitespace-nowrap md:hidden items-center justify-center"
      >
        <button
          className=" w-full h-full font-bold rounded-lg    text-secondary flex "
          onClick={[Function]}
          type="button"
        >
          <svg
            className="h-12 w-12"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 6h16M4 12h16M4 18h16"
            />
          </svg>
        </button>
      </div>
    </div>
  </nav>
  <div
    className="hidden"
  >
    <div
      className="flex felx-row bg-primary"
    >
      <ul
        className=" px-5 flex flex-col"
      >
        <li
          className="pb-1"
        >
          <button
            className="mr-3 text-gray-300 hover:text-highlight"
            onClick={[Function]}
            onMouseEnter={[Function]}
            type="button"
          >
            Home
          </button>
        </li>
        <li
          className="pb-1"
        >
          <button
            className="mr-3 text-gray-300 hover:text-highlight"
            onClick={[Function]}
            onMouseEnter={[Function]}
            type="button"
          >
            Contact
          </button>
        </li>
        <li
          className="pb-1"
        >
          <button
            className="mr-3 text-gray-300 hover:text-highlight"
            onClick={[Function]}
            onMouseEnter={[Function]}
            type="button"
          >
            Adopt
          </button>
        </li>
        <li
          className="pb-1"
        >
          <button
            className="mr-3 text-gray-300 hover:text-highlight"
            onClick={[Function]}
            onMouseEnter={[Function]}
            type="button"
          >
            Resources
          </button>
        </li>
      </ul>
      <span>
        <span
          className="hidden"
        >
          <button
            onClick={[Function]}
            onMouseEnter={[Function]}
            type="button"
          >
            <img
              alt="Website logo "
              className="w-16 h-16  cursor-pointer  rounded-full   "
              src="./profile.png"
            />
          </button>
        </span>
        <span
          className="block mt-7 sm:inline-block sm:mt-0 "
        >
          <button
            className="mr-3.5 w-28 h-10 text-highlight border-2 border-highlight rounded-lg "
            onClick={[Function]}
            onMouseEnter={[Function]}
            type="button"
          >
            Login
          </button>
          <button
            className=" w-28 h-10 text-primary bg-highlight hover:bg-darken rounded-lg "
            onClick={[Function]}
            onMouseEnter={[Function]}
            type="button"
          >
            Sign up
          </button>
        </span>
        <span
          className="bg-secondary text-sm text-white leading-none  rounded-full inline-flex mt-6 mb-4 md:mb-0 sm:mt-0 ml-24  sm:ml-14"
        >
          <button
            className="inline-flex items-center transition-colors duration-300 ease-in focus:outline-none hover:text-black focus:text-primary foucs:bg-primary rounded-l-full px-4 py-2 active"
            type="button"
          >
            <span>
              en
            </span>
          </button>
          <button
            className="inline-flex items-center transition-colors duration-300 ease-in focus:outline-none hover:text-black focus:text-primary foucs:bg-primary rounded-l-full px-4 py-2"
            type="button"
          >
            <span>
              ku
            </span>
          </button>
        </span>
      </span>
    </div>
  </div>
</div>
`;
